utils.py

```diff
@@ -888,13 +888,24 @@ def check_auth(user, hutch):
             return True
     return False
 
+# checks if ioc is marked as toggleable between defined versions of IOCs
+# schema will be ioc_name:permittedversion1,permittedversion2,etc
 def check_special(ioc, hutch):
-    lines = open(SPECIAL_FILE % hutch).readlines()
-    lines = [l.strip() for l in lines]
-    for l in lines:
-        if l == ioc:
+    with open(SPECIAL_FILE % hutch) as fp:
+        lines = fp.readlines()
+        lines = [l.strip()for l in lines]
+        for entry in lines:
+          ioc_vers_list = entry.split(":")
+          # case in which we have specified toggleable versions, check if toggleable version was chosen
+          if len(ioc_vers_list) > 1:
+            perm_version = ioc_vers_list[-1].split(",")
```

Comment:
You are still not looking at ioc_vers_list[0].  And if version is (the presumable default) "no_upgrade", there is no need to scan through the list at all.